itou/utils/management/commands/s3_antivirus_scan_resume_files.py

```diff
@@ -0,0 +1,60 @@
+import os
+import time
+
+import pyclamd
+import requests
+from django.core.management.base import BaseCommand, CommandError
+from tqdm import tqdm
+
+from itou.job_applications.models import JobApplication
+
+
+class Command(BaseCommand):
+    help = "Run ClamAV antivirus scan on resume files hosting in S3 storage type"
+
+    def handle(self, *args, **options):
+        # check that clamav is deployed in the current environment
+        if os.environ.get("CC_CLAMAV", False):
+            infected_files = []
+            self.stdout.write("Scan starts..")
+            job_applications = JobApplication.objects.exclude(resume_link__isnull=True).exclude(resume_link__exact="")
```

Comment:
pourquoi ne pas mettre les deux dans le meme exclude ?

Reply:
ah ? pour moi chainer deux excludes ou les mettre ensemble faisait un AND de la meme manière. Pour moi si on veut un OR il faut l'objet Q, enfin c'et mon réflexe. Sinon je suis en train d'apprendre un truc !

Reply:
Eh  ben, 10 ans de Django... Merci :sweat_smile: je ne sais pas pourquoi je n'ai jamais utilisé ce pattern. Y a pas de piège ? Quand tu ids "interprété de manière séquentielle", cela signifie-t-il que dans certains cas l'ordre des `exclude` ou `filter` compte ?

Reply:
A ma connaissance non : dire `je veux les balles pas jaunes et pas rouges` c'est pareil que `pas rouges et pas jaunes`.

Pour détailler plus le fonctionnement : 
- Chaque `.filter()` s'additionne. 
  -> `qs.filter(C1).filter(C2) == qs.filter(Q(C1) & Q(C2)`
- On peut traduire exclude en filter : `qs.exclude(C) == qs.filter(~Q(C))` donc ça s'additionne aussi.
  -> `qs.filter(C1).exclude(C2) == qs.filter(Q(C1) & ~Q(C2)`
- Idem pour ce qui est à l’intérieur de filter.
  -> `qs.filter(C1, C2) == qs.filter(Q(C1) & Q(C2))`
- Par contre attention aux éléments à l'intrieur d'un `exclude()`, la négation (`~`) inverse aussi l'opération d’addition. 
  `qs.exclude(C1, C2) == qs.filter(~(Q(C1) & Q(C2))) == qs.filter(~Q(C1) | ~Q(C2))`
  ce qui est différent de `qs.exclude(C1).exclude(C2) == qs.filter(~Q(C1) & ~Q(C2))` 
- Les opération `&` et `|` sont symétriques, donc tu peux inverser autant que tu veux l'ordre des filter/exclude.